api/app/db/crud/api_access_audits.py

```diff
@@ -0,0 +1,28 @@
+""" CRUD create function for audit logs
+"""
+from app.time_utils import get_utc_now
+import logging
+from sqlalchemy.exc import IntegrityError
+from app.db.models import APIAccessAudit
+
+import app.db.database
+
+logger = logging.getLogger(__name__)
+
+
+def create_api_access_audit_log(
+        username: str,
+        success: bool,
+        path: str) -> APIAccessAudit:
+    """ Create an audit log. """
+    with app.db.database.get_write_session_scope() as session:
+        now = get_utc_now()
+        audit_log = APIAccessAudit(create_user=username, path=path, success=success,
+                                   create_timestamp=now)
+        try:
+            session.add(audit_log)
+            session.commit()
+        except IntegrityError:
+            logger.info('Skipping duplicate record for username=%s @ %s for path=%s',
```

Comment:
I'm curious to know if this error handling is in here because it's an error that's already been encountered?

Reply:
Good point, maybe better off just letting this exception happen. There shouldn't be integrity issues - there's no unique key other than the id, so this shouldn't happen?

Reply:
I added this because there was a pattern of it's usage in other database creation calls but not sure this will ever happen because the only uniqueness constraint is the auto-incrementing primary key.

Reply:
I think drop it - let it raise the exception, let's see if it ever happens.

Reply:
if the other database creation calls you're talking about are related to the Env Canada models (and maybe also in our DB writes for observations and forecasts...?), then yes IntegrityError handling is required there because it's a frequent situation that we pull the same data from external sources multiple times - there's no way to know in advance whether the data is a duplicate or if it's been revised. But I don't think IntegrityError should happen here, unless for some reason there's multiple auth calls being made for the same user at the exact same timestamp??

Reply:
@andrea-williams Yes. @Sybrand Will do.